model/src/main/kotlin/utils/SortedSetConverter.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.utils
+
+import com.fasterxml.jackson.databind.util.StdConverter
+
+class SortedSetConverter<T : Comparable<T>> : StdConverter<Set<T>, Set<T>>() {
```

Comment:
For now, this is a generic implementation that works on `Comparable<T>`, which has the downside that the classes still need to implement `Comparable`, and the current implementations are confusing as they're not in line with `equals()`.

So maybe we should prefer to have dedicated converters per class that we want to sort as these could hard-code the property to sort on then, like

class SortedSetConverter : StdConverter<Set<Package>, Set<Package>>() {
    override fun convert(value: Set<Package>): Set<Package> = value.toSortedSet(compareBy { it.id })
}

That would allow us to remove the `Comparable` implementations in exchange for several more converted classes. But as there are only ~10 implementations that we need to care about as they influence serialization, that's maybe still a good idea.

Reply:
> So maybe we should prefer to have dedicated converters per class [...] That would allow us to remove the `Comparable` implementations in exchange for several more converted classes.

That's what I did now.